numpy/core/src/umath/loops.c.src

```diff
@@ -2720,46 +2722,247 @@ OBJECT_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUS
 }
 /**end repeat**/
 
+
+
+NPY_NO_EXPORT int
+_is_float_complex(PyObject *op)
+{
+    if (PyFloat_Check(op) || PyComplex_Check(op)) {
+        return 1;
+    }
+    return 0;
+}
+
+/**begin repeat
+ * #func = isnan, isinf, isfinite, signbit, logical_not, arccos, arccosh,
+           arcsin, arcsinh, arctan, arctanh, cos, sin, tan, cosh, sinh, tanh,
+           exp, expm1, log, log10, log1p, sqrt, ceil, trunc, fabs, floor,
+           arctan2, conjugate, square, reciprocal, _ones_like, sign, degrees,
+           rad2deg, radians, deg2rad, exp2, log2, cbrt, rint, spacing#
+   #out_bool = 1, 1, 1, 1, 1, 0*37#
+ */
 NPY_NO_EXPORT void
-OBJECT_sign(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
+OBJECT_@func@(char **args, npy_intp *dimensions, npy_intp *steps, void *data)
 {
-    PyObject *zero = PyLong_FromLong(0);
+    PyObject *ret;
+    PyArrayObject *arr;
+    static PyObject *ufunc=NULL, *fallback=NULL, *attr=NULL;
 
-    UNARY_LOOP {
+    /* variables for UNARY_LOOP */
+    char *ip1 = args[0], *op1 = args[1];
+    npy_intp is1 = steps[0], os1 = steps[1];
+    npy_intp n = dimensions[0];
+    npy_intp i;
+
+    npy_cache_import("numpy", "@func@", &ufunc);
+    if (ufunc == NULL) {
+        return;
+    }
+    npy_cache_import("numpy.core._objectmath", "@func@", &fallback);
+    if (fallback == NULL) {
+        return;
+    }
```

Comment:
Can we put these cache imports, and/or the function name, in `data` instead? Then we'd only need one loop function, rather than one for each ufunc (or perhaps two - one for boolean return values)